torch/csrc/jit/passes/prepack_folding.cpp

```diff
@@ -0,0 +1,66 @@
+#include <stack>
+
+#include <torch/csrc/jit/api/module.h>
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/passes/prepack_folding.h>
+
+namespace torch {
+namespace jit {
+
+// Must run this pass after constant folding.
+void FoldPrePackingOps(
+    script::Module& m,
+    const PrePackingOpsFilterFn& is_foldable_op,
+    const std::string& attr_prefix) {
+  int64_t uid = 0;
+  auto method = m.get_method("forward");
+  auto graph = method.graph();
+  std::stack<Block*> blocks_to_visit;
+  std::unordered_set<Node*> nodes_to_delete;
+  blocks_to_visit.push(graph->block());
+  std::string attr_name_base = attr_prefix + "._jit_pass_packed_weight_";
+  while (!blocks_to_visit.empty()) {
+    Block* b = blocks_to_visit.top();
+    blocks_to_visit.pop();
+    for (Node* n : b->nodes()) {
+      if (is_foldable_op(n)) {
+        auto optional_outputs = runNodeIfInputsAreConstant(n);
+        if (optional_outputs) {
+          auto outputs = optional_outputs.value();
+          TORCH_CHECK(outputs.size() == 1, "Prepack ops have single output");
+          auto attr_name = attr_name_base + c10::to_string(uid++);
+          TORCH_CHECK(
+              !(m.type()->findAttributeSlot(attr_name)),
+              "Attribute name ",
+              attr_name,
+              " already exist in",
+              " module of type:",
+              m.type()->name()->qualifiedName(),
+              ". Please make sure that",
+              " FoldPrePackingOps is run at the top level module only.");
```

Comment:
I think you should increment uid until the attribute is not present, see https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/quantization.cpp#L1733

Reply:
I am not follow why that should be done here? The code you pointed to seems to add a child module instance to some parent module. And from my understanding of the code you can try to add multiple child module to the same instance of parent module and thus you need to continue to increment.
This pass is little different in that we want to process this module once only and it should be the top level module. Thus if you try to run this pass on the top level module again and for some reason you come across this condition, then something has gone wrong and you should rather break than continue. Correct me if my understanding is wrong.